# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [15]:
library(tidyverse) 


# Assignment 1

First, we generate some random data:

In [16]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1         V2        V3        V4        V5        V6        V7       
1 0.01466672 0.7301251 0.4642758 0.1491614 0.6654272 0.8643277 0.7026997
2 0.45001897 0.8244432 0.4219221 0.5814999 0.4148736 0.4074916 0.5510216
  V8        V9         V10       V11       V12       V13       V14       
1 0.6205378 0.06747797 0.4021493 0.5128329 0.1448188 0.1373986 0.08865432
2 0.8180752 0.29776263 0.8681355 0.4866243 0.2288788 0.1224486 0.60943347
  V15       V16       V17       V18        V19       V20      
1 0.3778033 0.4278404 0.2417926 0.03931201 0.6086253 0.5803303
2 0.9593334 0.8385367 0.4483320 0.26494395 0.1197907 0.8039557

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [21]:
df2 <- df1 %>%
gather(key = 'Key', value = 'Value', V1:V20)
  

head(df2)

Key Value     
1 V1  0.01466672
2 V1  0.45001897
3 V2  0.73012510
4 V2  0.82444319
5 V3  0.46427577
6 V3  0.42192213

## Assignment 2

We will generate some data:

In [18]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [19]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [20]:
df5 <- full_join(df3, df4, by = 'id')
df5

Joining, by = "id"


id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook